cranelift-codegen/src/isa/arm64/inst.rs

```diff
@@ -1124,7 +1119,13 @@ fn arm64_get_regs(inst: &Inst) -> InstRegUses {
         &Inst::Nop | Inst::Nop4 => {}
     }
 
-    debug_assert!(iru.modified.is_empty());
+    // Enforce the invariant that if a register is in the 'modify' set, it
+    // should not be in 'defined' or 'used'.
+    iru.defined.remove(&iru.modified);
+    iru.used.remove(&Set::from_vec(
+        iru.modified.iter().map(|r| r.to_reg()).collect(),
+    ));
```

Comment:
Should this assert instead of modify?

Reply:
No; this new code correctly adjusts the three sets to make them be in accordance with what the regalloc.rs interface requires.